csrc/multidevice/lower_communication.cpp

```diff
@@ -139,18 +140,19 @@ CommParams createParamsForGatherScatter(
 // Adds one or zero Scatter communication to the vector 'comms'
 void lowerToScatter(
     DeviceIdxType my_device_index,
-    const DeviceMesh& sender_mesh,
-    const DeviceMesh& receiver_mesh,
+    TensorView* input_tv,
+    TensorView* output_tv,
     at::Tensor input_tensor,
     at::Tensor output_tensor,
     std::vector<std::shared_ptr<Communication>>& comms) {
   // we arbitrarily choose the first device of the sender mesh to be the root
-  auto root = sender_mesh.vector().at(0);
+  auto receiver_mesh = output_tv->getDeviceMesh();
```

Comment:
nit: `const auto&` here and everywhere it is a DeviceMesh, so we avoid copying a vector